HTTPServer.driver/PutCriteria.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HTTPServer.core;
+
+namespace HTTPServer.app
+{
+    public class PutCriteria : ICriteria
+    {
+        public bool ShouldRun(Request request)
+        {
+            return request.Method.Equals("PUT") && request.HttpVersion.Equals("HTTP/1.1") && !request.Uri.Equals("/");
```

Comment:
think of the "HTTP 1.1 only" restriction as another server setting (since version 1.0 is still a thing) so you're purposely saying that _this specific server_ should only respond to version 1.1.  you can still keep the independence and make the server ignorant if you think of your handler interface as a section of a pipeline and think of the version check as a "filter" in that pipeline.